### PR TITLE
fixes an issue with resolving IDs

### PR DESCRIPTION
**Pull request by johnspackman**
fixes an issue with resolving IDs, where if the object does not have any owner its absolute ID will never be found, even if it has an ID and is registered as top level;

also validation of ids and comment fixes

### CODE REVIEW

**Review by hkollmann — DISMISSED** [review]

File: uitests/objectid/source/class/objectid/Application.js [hunk 1]
```diff
@@ -53,6 +53,14 @@ qx.Class.define("objectid.Application", {
       this.getObject("exampleEditor/ctlrTitle").setModel(model);
 
       root.add(this.getObject("exampleEditor/container"), { left: 100, top: 250 });
+      
+      var button_panel = new qx.ui.toolbar.ToolBar();
+      button_panel.add(new qx.ui.toolbar.Button("Push Me"));
+      root.add(button_panel, { left: 100, top: 300 });
+      qx.core.Id.getInstance().register(button_panel, "buttons");
+      button_panel.setObjectId("buttons");
```
Comment (cboulanger) [inline uitests/objectid/source/class/objectid/Application.js]: Wouldn't it make sense to set the object id implicitly when registering the object? Specifying it twice seems redundant and might be a source of bugs. It would also save one line of code.

Comment (johnspackman) [reply uitests/objectid/source/class/objectid/Application.js]: the idea is that you can register shortcuts, so you could register A and C as top level, and then this code would be true: `qx.core.Id.getObject(“A/B/C/D/E”) === qx.core.Id.getObject(“C/D/E”)`
but when registering an object globally, you probably want to give it a different ID in that context, eg something more meaningful at a global level - so you have the option to override the ID for just globally, eg:
```
 qx.core.Id.register(C, “currentUser”);
/* .. snip .. */
qx.core.Id.getObject(“currentUser/edtName”).setValue(“George”);
qx.core.Assert.assertTrue (qx.core.Id.getObject("A/B/C") === qx.core.Id.getObject("currentUser"));
```
in other words, as the global objects set via `qx.core.Id.register()` are pointers into the ID heirarchy, the fact that you are registering something should not change the thing that you are registering

also, you could express your code as this:
```
       button_panel.setObjectId("buttons");
      qx.core.Id.getInstance().register(button_panel);
```
and `register()` will use the object’s objectId rather than assuming you need to override.

**Review by cboulanger — DISMISSED** [review]

**Review by hkollmann — DISMISSED** [review]

**Review by cboulanger — DISMISSED** [review]

File: framework/source/class/qx/core/MObjectId.js [hunk 2]
```diff
@@ -63,16 +64,26 @@ qx.Mixin.define("qx.core.MObjectId", {
...
     _applyOwner : function(value, oldValue) {
       if (!this.__changingOwner) {
         throw new Error("Please use API methods to change owner, not the property");
       }
     },
+    
+    /**
+     * Transform/validate objectId
+     */
+    _transformObjectId: function(value) {
```
Comment (derrell) [inline framework/source/class/qx/core/MObjectId.js]: This method isn't actually doing any transforms. Would it not be cleaner to have a customer `check` function that checked for string and no slashes, than to do it in a non-transforming `transform` function?

**Review by derrell — CHANGES_REQUESTED** [review]
My comment is solely a readability fix. If there's reason to do it as currently implemented, I'd be fine with it.

File: framework/source/class/qx/core/MObjectId.js (hunk 2 above)
Comment (johnspackman) [reply framework/source/class/qx/core/MObjectId.js]: yes i agree - funny, i've never used that feature before!

**Review by derrell — APPROVED** [review]

**Review by hkollmann — APPROVED** [review]

**Review by cboulanger — APPROVED** [review]

**Review by oetiker — APPROVED** [review]
ship it!

**Review by level420 — APPROVED** [review]

**Comment by cboulanger** [comment]
Four approvals, merging, thank you @johnspackman !
